tools/testing/test_selections.py

```diff
@@ -133,50 +137,121 @@ def _query_changed_test_files() -> List[str]:
     return lines
 
 
+def _get_previously_failing_tests() -> Set[str]:
+    PYTEST_FAILED_TESTS_CACHE_FILE_PATH = Path(".pytest_cache/v/cache/lastfailed")
+
+    if not PYTEST_FAILED_TESTS_CACHE_FILE_PATH.exists():
+        print(f"No pytorch cache found at {PYTEST_FAILED_TESTS_CACHE_FILE_PATH}")
+        print(f"cwd is {os.getcwd()}")
+        return set()
+
+    with open(PYTEST_FAILED_TESTS_CACHE_FILE_PATH, "r") as f:
+        last_failed_tests = json.load(f)
+
+    prioritized_tests = _parse_prev_failing_test_files(last_failed_tests)
+    return _python_test_file_to_test_name(prioritized_tests)
+
+
+def _parse_prev_failing_test_files(last_failed_tests: Dict[str, bool]) -> Set[str]:
+    prioritized_tests = set()
+
+    # The keys are formatted as "test_file.py::test_class::test_method[params]"
+    # We just need the test_file part
+    for test in last_failed_tests:
+        parts = test.split("::")
+        if len(parts) > 1:
+            test_file = parts[0]
+            prioritized_tests.add(test_file)
+
+    return prioritized_tests
+
+
+def _get_modified_tests() -> Set[str]:
+    try:
+        changed_files = _query_changed_test_files()
+    except Exception:
```

Comment:
Would be good to warn here
```suggestion
    except Exception as e:
    warnings.warn(f"Can't query changed test files due to {e}")
```